### PR TITLE
[promtail] Render pipelineStages with tpl function

### PR DESCRIPTION
**Pull request by chrizel**
Implements the suggestion by #2412 which allows us to work around the `toYaml` problems described under #2428

### CODE REVIEW

**Review by zanhsieh — CHANGES_REQUESTED** [review]
@chrizel 
Thank for the contribution. Can you bump version in Chart.yaml so the CI can merge this PR once approved?

**Review by zanhsieh — APPROVED** [review]

**Comment by conorevans** [comment]
As discussed [here](https://github.com/grafana/helm-charts/issues/2428#issuecomment-1667405261) this breaks any Go templating... you should consider reverting this. More importantly, you should also consider your semver - this is not a bugfix, it is a change in behaviour, which warrants a minor version bump. I mean at present it's a breaking change but even if it worked, it would be a minor change in behaviour.

**Pull request merged**
